internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1318,6 +1319,18 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 							Type:     pluginsdk.TypeBool,
 							Optional: true,
 						},
+						"revisions": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Computed: true,
+							MinItems: 1,
+							MaxItems: 2,
+							Set:      set.HashStringIgnoreCase,
```

Comment:
Does this need to be a set?

Reply:
No need, but Set feels more suitable in this case to stop users from inputting duplicates, can update to List if preferred.

Reply:
We only use sets if the API doesn't preserve the ordering. 

Regarding duplicates, you could add some validation in the expand function to check for duplicates and can error at apply time if that helps?